src/main/java/com/github/dockerjava/api/model/Device.java

```diff
@@ -45,6 +51,69 @@ public String getPathOnHost() {
         return pathOnHost;
     }
 
+    /**
+     * @link https://github.com/docker/docker/blob/6b4a46f28266031ce1a1315f17fb69113a06efe1/runconfig/opts/parse_test.go#L468
+     */
+    @Nonnull
+    public static Device parse(@Nonnull String deviceStr) {
+        String src = "";
+        String dst = "";
+        String permissions = "rwm";
+        final String[] arr = deviceStr.split(":");
+        switch (arr.length) {
+            case 3: {
+                permissions = arr[2];
+            }
+            case 2: {
+                if (validDeviceMode(arr[1])) {
+                    permissions = arr[1];
+                } else {
+                    dst = arr[1];
+                }
+            }
+            case 1: {
+                src = arr[0];
+                break;
+            }
+            default: {
+                throw new IllegalArgumentException("invalid device specification: " + deviceStr);
+            }
+        }
+
+        if (isEmpty(dst)) {
+            dst = src;
+        }
+
+        return new Device(permissions, dst, src);
+    }
+
+    /**
+     * ValidDeviceMode checks if the mode for device is valid or not.
+     * Valid mode is a composition of r (read), w (write), and m (mknod).
+     *
+     * @link https://github.com/docker/docker/blob/6b4a46f28266031ce1a1315f17fb69113a06efe1/runconfig/opts/parse.go#L796
+     */
+    private static boolean validDeviceMode(String deviceMode) {
+        Map<String, Boolean> validModes = new HashMap<>(3);
+        validModes.put("r", true);
+        validModes.put("w", true);
+        validModes.put("m", true);
+
+        if (isEmpty(deviceMode)) {
+            return false;
+        }
+
+        for (char ch : deviceMode.toCharArray()) {
+            final String mode = String.valueOf(ch);
+            if (isNotTrue(validModes.get(mode))) {
+                return false; // wrong mode
+            }
+            validModes.put(mode, false);
```

Comment:
OK. Now I got it. So it must be declared within the method then. Sorry for bothering.